examples/gno.land/r/gnoland/valopers/valopers.gno

```diff
@@ -99,6 +100,14 @@ func Register(cur realm, moniker string, description string, address_XXX std.Add
 
 	// TODO add address derivation from public key
 	// (when the laws of gno make it possible)
+	matched, err := keyderive.AddressMatchesKey(address_XXX.String(), pubKey)
```

Comment:
This check consumes a lot of gas. In my local tests, calling `Register` without `AddressMatchesKey` has gas used 12640012. With this check it has gas used 20245386. That's an increase of 60%. Is this wanted?

Reply:
Hello Jeff,
thanks for testing this an give your results, I think this is explained by the cryptography we're applying to compute the address from the publicKey. Also, because we're using a golang Native lib sha256 ?

The code I added is not that complex  Iam mostly using pre-existent libs or copy-pasting functions we do on tm2 package, ex: [secp256k1 key to bech32](https://github.com/gnolang/gno/blob/34ba1efdb4fe0e23183a3524ff306a0429de9928/tm2/pkg/crypto/secp256k1/secp256k1.go#L122-L130).

But maybe we can find a way to optimize one of the packages

Reply:
Sha256 shouldn't use much gas, if any at all.

@Villaquiranm is this because of ripemd160, or because of bech32?

Can you run some tests and verify?

You can probably get a rough first idea using `gno test -print-runtime-metrics`, though it'd be better to use actual gas